gen/gen.go

```diff
@@ -684,15 +684,25 @@ func (g *Generator) DefaultValue(member ParameterType, isDocString bool) string
 	// Cases that may have member.Default
 	case strings.HasPrefix(member.Type, "enum."):
 		if member.Default == nil {
-			if member.Type == "enum.optional_bool" {
-				// This Undefined is a special one that is not the zero-value, so that
-				// a stdbool.h bool cast correctly to WGPUOptionalBool; this means we
-				// must explicitly initialize it
-				return ref("WGPUOptionalBool_Undefined")
-			} else if isDocString {
-				return "(@ref " + g.CType(member.Type, "") + ")0"
+			_, name, _ := strings.Cut(member.Type, ".")
+
+			// Find the enum type.
+			idx := slices.IndexFunc(g.Enums, func(e Enum) bool { return e.Name == name })
+			if idx == -1 {
+				panic("Invalid enum type: " + name)
+			}
+
+			// If the enum type has an explicit "Undefined" use it, otherwise, use 0.
+			enumType := g.Enums[idx]
+			undefIdx := slices.IndexFunc(enumType.Entries, func(entry *EnumEntry) bool { return entry != nil && entry.Name == "undefined" })
+			if undefIdx == -1 {
+				if isDocString {
+					return "(@ref " + g.CType(member.Type, "") + ")0"
+				} else {
+					return "_wgpu_ENUM_ZERO_INIT(" + g.CType(member.Type, "") + ")"
+				}
 			} else {
```

Comment:
We have four enums where `Undefined` is 1 instead of 0 (because 0 is `BindingNotUsed`).

Maybe instead of special casing the name `Undefined` we should special case the value 0?
Or special case only when the name is `Undefined` AND the value is 0?

Reply:
I don't think it should matter what `Undefined` is actually equal? Either we have an explicit default in the yaml for if we don't want `Undefined` to be the default, or whatever the value of `Undefined` is should be the default. 

FWIW, the enums you're referring to, (`WGPUBufferBindingType`, `WGPUSamplerBindingType`, `WGPUTextureSampleType`, and `WGPUStorageTextureAccess`), all already assume their defaults should be `Undefined` when calling their `INIT` explicitly. The only time that they use `BindingNotUsed` instead is when they are being `INIT` via their wrapping `WGPUBindGroupLayoutEntry`, but that `INIT` macro explicitly uses a zero init for the wrapped structs.

Anyways, I don't think the additional condition is necessary? Especially if the name of the enum is `Undefined`.

Reply:
> FWIW, the enums you're referring to, (`WGPUBufferBindingType`, `WGPUSamplerBindingType`, `WGPUTextureSampleType`, and `WGPUStorageTextureAccess`), all already assume their defaults should be `Undefined` when calling their `INIT` explicitly. The only time that they use `BindingNotUsed` instead is when they are being `INIT` via their wrapping `WGPUBindGroupLayoutEntry`, but that `INIT` macro explicitly uses a zero init for the wrapped structs.

Ahh true. I forgot how this was set up.